x-pack/plugins/apm/public/components/shared/charts/TransactionCharts/index.tsx

```diff
@@ -149,79 +139,92 @@ export class TransactionCharts extends Component<TransactionChartProps> {
       </EuiFlexItem>
     );
   }
+  const { responseTimeSeries, tpmSeries } = charts;
+  const { transactionType } = urlParams;
+  const maxY = getMaxY(responseTimeSeries);
+  let formatter = getDurationFormatter(maxY);
```

Comment:
This is now stateful afaict. Relying on side effects like `onToggleLegend` won't ensure that the component re-renders. I think instead you should make `formatter` explicitly stateful:

```ts
  const [visibleSeries, setVisibleSeries] = useState(responseTimeSeries)`;
  const maxY = getMaxY(visibleSeries);
  const formatter = getDurationFormatter(maxY);

  function onToggleLegend(visibleSeries: TimeSeries[]) {
    setVisibleSeries(visibleSeries)
  }
```

Reply:
I wouldn't say that it is stateful, it only changes the formatter reference when `onToggleLegend` is called. But I'm fine doing as you suggested, it is more clear what's happening. Thanks for the suggestion.

Reply:
As I wrote somewhere else, this currently only works because the component _does_ re-render. If it didn't re-render the change to the `formatter` identify wouldn't have any effect. As an example, try this:

```
setTimeout(() => {
 formatter = getDurationFormatter(10);
}, 5000)
```

After 5 seconds the formatter should be updated. But it won't have any effect on the chart unless it's re-rendered.